drivers/pipes/pipe_common.h

```diff
@@ -120,6 +120,7 @@ struct pipe_dev_s
   pipe_ndx_t d_wrndx;       /* Index in d_buffer to save next byte written */
   pipe_ndx_t d_rdndx;       /* Index in d_buffer to return the next byte read */
   pipe_ndx_t d_bufsize;     /* allocated size of d_buffer in bytes */
+  pipe_ndx_t d_pollthres;   /* Buffer threshold for POLLIN to occur */
```

Comment:
should we support POLLOUT direction too

Reply:
Yes, after some thinking, I think it is necessary.
We should avoid writing when only preamble can be written.